pkg/utils/string_utils_test.go

```diff
@@ -45,6 +45,27 @@ func TestSplitStringByDelimiter(t *testing.T) {
 			expected:  []string{"foo:", "!env", "FOO"},
 			expectErr: false,
 		},
+		{
+			name:      "Single quoted value with nested double quotes",
+			input:     "core '.security.users[\"github-dependabot\"].access.key.id'",
+			delimiter: ' ',
+			expected:  []string{"core", ".security.users[\"github-dependabot\"].access.key.id"},
+			expectErr: false,
```

Comment:
Can you maybe generate more tests, especially for edge cases? Such as if somebody nests single quotes inside of a single quote.